_observing-your-data/prometheusmetrics.md

```diff
@@ -0,0 +1,125 @@
+    "connector": "prometheus",
+    "properties" : {
+        "prometheus.uri" : "http://localhost:8080",
+        "prometheus.auth.type" : "awssigv4",
+        "prometheus.auth.region" : "us-east-1",
+        "prometheus.auth.access_key" : "{{accessKey}}"
+        "prometheus.auth.secret_key" : "{{secretKey}}"
+    }
+}]
+```
+
+After configuring Prometheus in the `datasources.json` file, run the following command to load the configuration into the OpenSearch keystore. The configuration is securely stored in the keystore, as it contains sensitive credential information.
+
+```
+bin/opensearch-keystore add-file plugins.query.federation.datasources.config datasources.json
+```
+
+If you are updating the keystore during runtime, refresh the keystore using following API command:
+
+```bash
+POST /_nodes/reload_secure_settings
+{
+  "secure_settings_password":""
+  
+}
+```
+{% include copy-curl.html %}
+
+Metrics collected from Prometheus are displayed in the **Observability** application under **Metrics analytics**. For further information, see the [Prometheus Connector](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/admin/prometheus_connector.rst) GitHub page.
+
+## Prometheus PPL queries
+
+You can define [Piped Processing Language]({{site.url}}{{site.baseurl}}/search-plugins/sql/ppl/index) (PPL) queries against metrics collected by Prometheus. The following example shows a metric-selecting query with specific dimensions:
+
+```
+source = my_prometheus.prometheus_http_requests_total | stats avg(@value) by span(@timestamp,15s), handler, code
+```
+
+## Creating visualizations based on metrics
+
+You can create visualizations based on metrics collected by your OpenSearch cluster in the in the new **Metrics Analytics** section of the **Observability** application in OpenSearch Dashboards. To create a visualization, choose the metrics you would like to include from the list under **Available Metrics**. These visualizations will be displayed in the **Metrics Analytics** section of **Observability** as seen in the following image. 
+
+![Metrics UI example 2]({{site.url}}{{site.baseurl}}/images/metrics/metrics2.png)
+
+After creating a visualization, you can save it with the following steps:
+
+1. Select **Save**.
+2. You are then prompted for a **Custom operational dashboards/application**.
+3. Choose one of the provided options.
+4. Enter name values under the **Metric Name** fields.
+5. Select **Save**.
+
+Additionally, you can create a custom visualization generated with a PPL query under [Event analytics]({{site.url}}{{site.baseurl}}/observing-your-data/event-analytics/) to fit your needs. The following image is an example of a visualization based on a PPL query.
+
+![Metrics UI example 3]({{site.url}}{{site.baseurl}}/images/metrics/metrics3.png)
+
+You can save the visualization with the following steps:
+
+1. Select **Save**.
+2. You are then prompted for a **Custom operational dashboards/application**.
```

Comment:
same edits as above.

Reply:
I would prefer that the save process is combined with the creation process, because I assume they wouldn't create a visualization unless they wanted to save it. You'd have to write steps for the create process, in that case.

Reply:
One is saving visualization from metric, while the other is saving from a PPL query. I have re-worded that to make it more clear